src/neubot/dash_impl.hpp

```diff
@@ -0,0 +1,163 @@
+    Settings settings;
+    int DASH_RATES[] = {100,  150,  200,  250,  300,   400,  500,
+                        700,  900,  1200, 1500, 2000,  2500, 3000,
+                        4000, 5000, 6000, 7000, 10000, 20000};
+    std::string url = "http://127.0.0.1/dash/download/";
+
+    if (iteration > DASH_MAX_ITERATION) {
+        transport->close([=]() { cb(NoError(), measurements); });
+        return;
+    };
+
+    // Bisect
+    while (DASH_RATES[rate_index] < speed_kbit) {
+        rate_index++;
+    }
+    rate_index -= 1;
+    if (rate_index < 0)
+        rate_index = 0;
+
+    int rate_kbit = DASH_RATES[rate_index];
+    int count = ((rate_kbit * 1000) / 8) * DASH_SECONDS;
+    url += std::to_string(count);
+    settings["http/url"] = url;
+    double saved_times = mk::time_now();
+
+    http::request_send(
+        transport, settings,
+        {
+            {"Authorization", auth},
+        },
+        "",
+        [=](Error error) {
+            if (error) {
+                std::cout << "Error: " << (int)error;
+                cb(error, nullptr);
+                return;
+            }
+
+            request_recv_response(
+                transport,
+                [=](Error error, Var<Response> res) {
+                    if (error) {
+                        std::cout << "Error: " << (int)error;
+                        cb(error, nullptr);
+                        return;
+                    }
+
+                    double new_times = mk::time_now();
+                    float length = res->body.length();
+                    double time_elapsed = new_times - saved_times;
+
+                    if (time_elapsed < 0) {
+                        std::cout << "Error: " << (int)error;
+                        cb(error, nullptr);
+                        return;
+                    }
+                    // TODO
+                    if (auth != "") {
+                        json::object_t result = {
+                            //{"connect_time", self.rtts[0]}
+                            //{"delta_user_time", delta_user_time}
+                            //{"delta_sys_time", delta_sys_time}
+                            {"elapsed", time_elapsed},        // OK
+                            {"elapsed_target", DASH_SECONDS}, // OK
+                            //{"internal_address", stream.myname[0]}
+                            {"iteration", iteration}, // OK
+                            //{"platform", sys.platform}
+                            {"rate", rate_kbit}, // OK
+                            //{"real_address", self.parent.real_address}
+                            //{"received", received}
+                            //{"remote_address", stream.peername[0]}
+                            //{"request_ticks", self.saved_ticks}
+                            //{"timestamp", mk::time_now}
+                            //{"uuid", self.conf.get("uuid")}
+                            //{"version", utils_version.NUMERIC_VERSION}
```

Comment:
This would be `MEASUREMENT_KIT_VERSION` defined in `measurement_kit/common.hpp`.